hack/test-cmd.sh

```diff
@@ -682,9 +682,7 @@ __EOF__
 
   ### Perform a rolling update with --image
   # Command
-  kubectl rolling-update frontend --image=kubernetes/pause --update-period=10ns --poll-interval=10ms "${kube_flags[@]}"
-  # Post-condition: current image IS kubernetes/pause
-  kube::test::get_object_assert 'rc frontend' '{{range \$c:=$rc_container_image_field}} {{\$c.image}} {{end}}' ' +kubernetes/pause +'
+  [ ! kubectl rolling-update frontend --image=kubernetes/pause --timeout=1s --update-period=10ns --poll-interval=10ms "${kube_flags[@]}" || true
```

Comment:
There aren't currently any negative tests around it in e2e, which isn't surprising since the main entrypoint for doing work in the e2e tests (`runKubectl`) is hard-coded to fail the test on non-zero exit codes. I'd need to add some sort of support for expected error codes. Should I do that as part of this PR?